src/lib/es2015.core.d.ts

```diff
@@ -43,7 +43,13 @@ interface Array<T> {
      */
     copyWithin(target: number, start: number, end?: number): this;
 
-    toLocaleString(locales: string | string[], options?: Intl.NumberFormatOptions & Intl.DateTimeFormatOptions): string;
+    /**
+     * Returns a string representation of an array.
+     * Each element is converted to a string using its `toLocaleString` method, then concatenated using the list separator of the current locale.
+     * @param locales Passed as the `locales` parameter to each array element's `toLocaleString` method.
+     * @param options Passed as the `options` parameter to each array element's `toLocaleString` method.
+     */
+    toLocaleString(locales?: string | readonly string[], options?: object): string;

```

Comment:
See notes towards the bottom of the PR description for discussion regarding the type of `options` here.

I'm ambivalent towards the change, but thought it would be worth proposing.